### PR TITLE
[ACS-7690] Saving adding o a new hold

### PR DESCRIPTION
**Pull request by DaryaBalvanovich**
**Please check if the PR fulfills these requirements**

https://hyland.atlassian.net/browse/ACS-7690

closed PR https://github.com/Alfresco/alfresco-ng2-components/pull/9807

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
no endpoint for create holds

**What is the new behaviour?**
integrate create and assign endpoints

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/src/lib/legal-hold/services/legal-holds.service.spec.ts [hunk 1]
```diff
@@ -116,4 +118,44 @@ describe('LegalHoldsService', () => {
...
+            service.createHold(mockId, mockHold).subscribe((holds) => {
+                expect(holds).toEqual(legalHoldEntry);
+                expect(service.legalHoldApi.createHold).toHaveBeenCalledWith(mockId, mockHold);
+                done();
+            });
+        });
+    });
+
+    describe('createHolds', () => {
+        it('should creaet list of holds', (done) => {
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply lib/content-services/src/lib/legal-hold/services/legal-holds.service.spec.ts:141]: nice catch, thanks

File: lib/content-services/src/lib/legal-hold/services/legal-holds.service.spec.ts [hunk 2]
```diff
@@ -116,4 +118,44 @@ describe('LegalHoldsService', () => {
...
+            const mockHold = [
+                {
+                    name: 'Hold 1',
+                    reason: 'reason 1'
+                }
+            ];
+            spyOn(service.legalHoldApi, 'createHold').and.returnValue(Promise.resolve(legalHoldEntry));
+
+            service.createHold(mockId, mockHold).subscribe((holds) => {
+                expect(holds).toEqual(legalHoldEntry);
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply lib/content-services/src/lib/legal-hold/services/legal-holds.service.spec.ts:133]: yes, sound good
renamed

File: lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts [hunk 3]
```diff
@@ -85,4 +85,26 @@ export class LegalHoldService {
     unassignHold(holdId: string, nodeId: string): Observable<void> {
         return from(this.legalHoldApi.unassignHold(holdId, nodeId));
     }
+
+    /**
+     * Create hold.
+     *
+     * @param filePlanId The identifier of a file plan. You can also use the -filePlan- alias.
+     * @param holds Array of one hold to create
+     * @returns List of created holds Observable<HoldEntry[]>
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts:94]: fixed

File: lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts [hunk 4]
```diff
@@ -109,4 +108,50 @@ export class LegalHoldApi extends BaseApi {
...
     }
+
+    /**
+     * Create new hold
+     *
+     * @param filePlanId The identifier of a file plan. You can also use the -filePlan- alias.
+     * @param holds Array of one hold
+     * @returns Promise<HoldEntry>
+     */
+    createHold(filePlanId: string, holds: Hold[]): Promise<HoldEntry> {
```
Comment (MichalKinas) [inline lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts:119]: Should be just single hold

File: docs/content-services/services/legal-hold.service.md [hunk 5]
```diff
@@ -17,7 +17,17 @@ Manages holds for nodes.
     Gets the list of holds for a node.
     -   _filePlanId_: `string` - The identifier of a file plan. You can also use the -filePlan- alias
     -   _options_: `ContentPagingQuery` - Optional parameters supported by JS-API
-    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]>` - List of holds
+    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]>` - List of holds <br/>
+-   **createHold**(filePlanId: `string`, holds: [`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]>`<br/>
+    Gets the list of holds for a node.
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: fixed

File: docs/content-services/services/legal-hold.service.md [hunk 6]
```diff
@@ -17,7 +17,17 @@ Manages holds for nodes.
...
+    -   _filePlanId_: `string` - The identifier of a file plan. You can also use the -filePlan- alias
+    -   _holds_: `<`[`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]>` - Array of one hold that should be created
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: refactored

File: docs/content-services/services/legal-hold.service.md [hunk 7]
```diff
@@ -17,7 +17,17 @@ Manages holds for nodes.
...
+    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`HoldEntry`](../../../lib/js-api/src/api/gs-core-rest-api/docs/HoldEntry.md)`>` - Hold entry<br/>
+-   **createHolds**(filePlanId: `string`, holds: [`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]>`<br/>
+    Gets the list of holds for a node.
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: done

File: lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts [hunk 8]
```diff
@@ -85,4 +85,26 @@ export class LegalHoldService {
...
+     */
+    createHold(filePlanId: string, holds: Hold[]): Observable<HoldEntry> {
```
[the remark this answers is not in the document]
Comment (DaryaBalvanovich) [reply lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts]: sounds good, refactored

File: lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts (hunk 4 above)
Comment (DaryaBalvanovich) [reply lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts]: done

File: docs/content-services/services/legal-hold.service.md [hunk 9]
```diff
@@ -17,7 +17,17 @@ Manages holds for nodes.
     Gets the list of holds for a node.
     -   _filePlanId_: `string` - The identifier of a file plan. You can also use the -filePlan- alias
     -   _options_: `ContentPagingQuery` - Optional parameters supported by JS-API
-    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]>` - List of holds
+    -   **Returns** [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]>` - List of holds <br/>
+-   **createHold**(filePlanId: `string`, holds: [`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)`[]`): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`HoldEntry`](../../../lib/js-api/src/api/gs-core-rest-api/docs/HoldEntry.md)`>`<br/>
```
Comment (MichalKinas) [inline docs/content-services/services/legal-hold.service.md:21]:
```suggestion
-   **createHold**(filePlanId: `string`, hold: [`Hold`](../../../lib/js-api/src/api/gs-core-rest-api/docs/Hold.md)): [`Observable`](http://reactivex.io/documentation/observable.html)`<`[`HoldEntry`](../../../lib/js-api/src/api/gs-core-rest-api/docs/HoldEntry.md)`>`<br/>
```

File: lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts [hunk 10]
```diff
@@ -61,35 +60,34 @@ export class LegalHoldApi extends BaseApi {
      * @param nodeIds one element list with id of nodes to assign to existing hold
      * @returns Promise<HoldEntry>
      */
-    assignHold(nodeIds: {id: string}[], holdId: string): Promise<HoldEntry> {
+    assignHold(nodeIds: { id: string }[], holdId: string): Promise<HoldEntry> {
```
Comment (MichalKinas) [inline lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts:63]:
```suggestion
    assignHold(nodeId: { id: string }, holdId: string): Promise<HoldEntry> {
```

Comment (MichalKinas) [inline lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts:63]: If docs for that method are already added they should be adjusted as well

Comment (DaryaBalvanovich) [reply lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts:63]: should be done in scope of https://github.com/Alfresco/alfresco-ng2-components/pull/9841
@tomson7777

Comment (DaryaBalvanovich) [reply lib/js-api/src/api/gs-core-rest-api/api/legal-hold.api.ts:63]: should be done in scope of https://github.com/Alfresco/alfresco-ng2-components/pull/9841
@tomson7777

File: docs/content-services/services/legal-hold.service.md (hunk 9 above)
Comment (DaryaBalvanovich) [reply docs/content-services/services/legal-hold.service.md]: fixed, thanks

**Review by MichalKinas — APPROVED** [review]

**Pull request merged**